doc/extensions/acrbuildext/azext_acrbuildext/build.py

```diff
@@ -255,20 +258,29 @@ def acr_queue(cmd,
             name, value = name_value.split('=', 1)
             build_arguments.append(BuildArgument(name, value, True))
 
-    try:
-        build_request = QuickBuildRequest(
-            source_location=source_location,
-            platform=platform,
-            docker_file_path=docker_file_path,
-            image_name=image_name,
-            is_push_enabled=is_push_enabled,
-            timeout=timeout,
-            build_arguments=build_arguments)
-
-        result = LongRunningOperation(cmd.cli_ctx)(client_registries.queue_build(
-            build_request=build_request, resource_group_name=resource_group_name, registry_name=registry_name))
-
-        size = os.path.getsize(tar_file_path)
+    build_request = QuickBuildRequest(
+        source_location=source_location,
+        platform=platform,
+        docker_file_path=docker_file_path,
+        image_name=image_name,
+        is_push_enabled=is_push_enabled,
+        timeout=timeout,
+        build_arguments=build_arguments)
+
+    result = LongRunningOperation(cmd.cli_ctx)(client_registries.queue_build(
+        build_request=build_request, resource_group_name=resource_group_name, registry_name=registry_name))
+
+    if isLocalFile:
+        try:
+            size = os.path.getsize(tar_file_path)
+        except OSError:
+            raise CLIError(
+                "Get the size of file {0} failed.".format(tar_file_path))
+        finally:
+            if os.path.exists(tar_file_path):
+                logger.debug(
+                    "Starting to delete the archived source code from '{}'.".format(tar_file_path))
+                os.remove(tar_file_path)
```

Comment:
You might need a try-except for the two `os` operations.

Reply:
os.path.exists does not raise any error. os.remove raises error when the argument is not a file. If tar_file_path exists, then it must be a file because of the way it is constructed.
tar_file_path = os.path.join(tempfile.gettempdir(),
                                 "source_archive_{}.tar.gz".format(hash(os.times())))

Reply:
The extension for tar_file_path is ".tar.gz", which means that it normally will be a file rather than anything else.

Reply:
`os.remove` could raise if the file is still used or by another process?

Reply:
......You win

Reply:
Do you want to introduce mutex or lock for the tar_file_path?

Reply:
Can you explain the possible scenario that another process happens to be using the same file? Since each time the file is generated, the name will be different. Other process will have their own tar_file_path if they want to use that and that file will be different from this one.

Reply:
Let me check how people normally deal with issues related to concurrencies when the file might be shared between different processes or threads. I cannot think of a situation where another process would like to read or write to the file.

Reply:
I tend to be extra careful when dealing with files. I would write this block as
```
try:
    os.remove(file)
except OSError:
    pass # you could further check if it's due to file doesn't exist and raise otherwise
except PermissionError
    raise CLIError('some message')
```
If you have to deal with concurrency, checking for existence doesn't guarantee you it will exist when you delete it. I admit it could be a rare case so it's up to you whether you want to handle it.

Reply:
Make sense. I will do  that. Thank you!!!